Add Astro 5.10 blog (#1622)
withastro/astro.build: src/content/blog/astro-5100.mdx

```astro
<Image
	src="/profile.jpg"
	alt="Profile photo"
	fit="cover"
	position="center top"
	width={300}
	height={300}
/>
```

### Upgrading from experimental responsive images

If you were using experimental responsive images, you should remove the experimental flag from your `astro.config.mjs` file:

```js title="astro.config.mjs" del={3} ins={6}
export default defineConfig({
	experimental: {
		responsiveImages: true,
	},
	image: {
		responsiveStyles: true,
	},
});
```

For complete details, see the [Images guide](https://docs.astro.build/en/guides/images/).

## Improvements to experimental Content Security Policy

Astro 5.9 shipped experimental support for Content Security Policy (CSP) meta tags, bringing powerful security features to your Astro sites. In 5.10, we've made several improvements based on community feedback. The most important is that we now support generating CSP headers, including for static sites.

On-demand rendered pages will now send Response headers for CSP, rather than meta tags. This improves performance in Chrome, and adds support for more directives, like `report-uri` and `frame-ancestors`, which are not supported in meta tags.

> By default prerendered pages will still use meta tags, but there is now core support for generating headers for prerendered pages as well. Eventually this will be implemented in all adapters, but currently only the `@astrojs/netlify` adapter supports it. To enable this, add the `experimentalStaticHeaders` option to your adapter configuration:

Reviewer: Should we mention the version of the adapter where `experimentalStaticHeaders` is available? The version is `6.4.0`

> By default prerendered pages will still use meta tags, but there is now core support for generating headers for prerendered pages as well. Eventually this will be implemented in all adapters, but currently only the `@astrojs/netlify` adapter supports it. To enable this, add the `experimentalStaticHeaders` option to your adapter configuration:

Reviewer: ```suggestion
By default, prerendered pages will still use meta tags, but there is now core support for generating headers for prerendered pages as well. Eventually, this will be implemented in all adapters, but currently only the `@astrojs/netlify` adapter supports it. To enable this, add the `experimentalStaticHeaders` option to your adapter configuration:
```

Do you think it's worth explaining that this feature can only be supported by certain adapters? Not sure how to phrase it, but for example, Node.js won't be able to support it. I would like to set correct expectations, as the current wording might mislead users and lead them to believe that even Node.js will receive this feature.

Reviewer: Ah yeah. Even a JSON file with our format could work. We need to map back the correct `RouteData` and it could work. Nice thinking!

---
title: 'Astro 5.10'

> description: 'Astro 5.10 brings responsive images for everyone, plus experimental live content collections, CSP improvements, and more'

Reviewer 2: ```suggestion
description: 'Astro 5.10 brings responsive images for everyone, plus experimental live content collections, CSP improvements, and more!'
```

## Responsive images are now stable

The Astro responsive images feature is now stable and ready for production!

Responsive images automatically generate optimized `srcset` and `sizes` attributes, and can generate the styles needed to ensure your images load quickly and display beautifully across all screen sizes. Say goodbye to layout shifts and slow-loading images that hurt your Core Web Vitals scores.

### Get started with responsive images

Most people should enable automatic styles for responsive images in your `astro.config.mjs` file:

```js title="astro.config.mjs" ins={3}
export default defineConfig({
	image: {
		responsiveStyles: true,
	},
});
```

You enable responsive images by setting a `layout`, either globally in your Astro config or on a per-image basis.

```js title="astro.config.mjs" {4}
export default defineConfig({
	image: {
		responsiveStyles: true,
		layout: 'constrained',
	},
});

> ```

Author: @sarah11918 I think combining them is good, but I don't think this fully captures it. I wonder if you can work out a succinct way of saying that `layout` tells Astro how the image will resize (so it can generate the right srcset/sizes etc for that behaviour) and the styles ensure that it *does* resize that like. It's about making sure they match up.

**Layout options:** Choose from `constrained`, `fixed`, or `full-width` layouts to control how your images behave:

```astro
<Image
	src="/hero.jpg"
	alt="A panoramic view of the mountains"
	layout="full-width"
/>
```

**Priority loading:** Use the new `priority` prop to optimize critical images that appear above the fold:

```astro
<Image src="/hero.jpg" alt="Hero image" priority />
```

> When you add `priority`, Astro automatically sets `loading="eager"`, `decoding="sync"`, and `fetchpriority="high"` to ensure the image loads immediately. Use this sparingly – ideally only one image per page should have priority loading. A good rule of thumb is to check if your [LCP element](https://web.dev/lcp/#lcp-element) is an image, and if so, set the `priority` flag on it.

Author: Good point!

> **Enhanced cropping controls:** The `fit` and `position` properties give you precise control over how images are cropped and positioned:

Author: Ah, another subtle point that I've not explained well! `fit` and `position` don't just control the styles, they control how the actual images are cropped too

			}
		} satisfies ExportedHandler<Env>,
		MyDurableObject,
	}
}
```

For more details, see the updated [Cloudflare adapter documentation](https://docs.astro.build/en/guides/integrations-guide/cloudflare#workerentrypoint).

Thanks to [Alexander Niebuhr](https://github.com/alexanderniebuhr) who contributed this feature, and does most of the work on the `@astrojs/cloudflare` adapter!

## Bug fixes

As always, we've been working hard on fixing issues since the [5.9 release](/blog/astro-590). See [the changelog](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md) for all the details.

## Community

The Astro core team is:

<Mention name="ben" />, <Mention name="caleb" />, <Mention name="chris" />,
<Mention name="ema" />, <Mention name="erika" />, <Mention name="florian" />,
<Mention name="fred" />, <Mention name="fuzzy" />, <Mention name="hideoo" />,
<Mention name="luiz" />, <Mention name="matt" />, <Mention name="matthew" />,
<Mention name="nate" />, <Mention name="reuben" />, <Mention name="sarah" />,
and <Mention name="yan" />.

Thanks to all the other contributors who helped make Astro 5.9 possible with code and docs additions and improvements, including:

> [Alexander Niebuhr](https://github.com/alexanderniebuhr), [Anshul Gupta](https://github.com/ansg191), [Armand Philippot](https://github.com/ArmandPhilippot), [benosmac](https://github.com/benosmac), [Han Seung Min - 한승민](https://github.com/arHSM), [Junseong Park](https://github.com/jsparkdev), [kato takeshi](https://github.com/tkskto), [knj](https://github.com/jp-knj), [liruifengv](https://github.com/liruifengv), [Martin Haug](https://github.com/reknih), [Martin Trapp](https://github.com/martrapp), [Nin3](https://github.com/Nin3lee), [Paul Valladares](https://github.com/dreyfus92), [Quinn Blenkinsop](https://github.com/qw-in), [Reuben Tier](https://github.com/TheOtterlord), [Thomas Bonnet](https://github.com/thomasbnt), and [zaitovalisher](https://github.com/zaitovalisher)

Reviewer 3: ```suggestion
[Alexander Niebuhr](https://github.com/alexanderniebuhr), [Anshul Gupta](https://github.com/ansg191), [Armand Philippot](https://github.com/ArmandPhilippot), [benosmac](https://github.com/benosmac), [Han Seung Min - 한승민](https://github.com/arHSM), [Junseong Park](https://github.com/jsparkdev), [kato takeshi](https://github.com/tkskto), [knj](https://github.com/jp-knj), [liruifengv](https://github.com/liruifengv), [Martin Haug](https://github.com/reknih), [Martin Trapp](https://github.com/martrapp), [Nin3](https://github.com/Nin3lee), [Paul Valladares](https://github.com/dreyfus92), [Quinn Blenkinsop](https://github.com/qw-in), [Thomas Bonnet](https://github.com/thomasbnt), and [zaitovalisher](https://github.com/zaitovalisher)
```

Just run it again locally with the latest version which exclude Reuben as mentioned above in the core team.

For more details, see the [experimental CSP documentation](https://docs.astro.build/en/reference/experimental-flags/csp/).

## Customizable Cloudflare Workers entrypoint

> Some Cloudflare Workers features, like [Durable Objects](https://developers.cloudflare.com/workers/runtime-apis/durable-objects/), [Cloudflare Queues](https://developers.cloudflare.com/queues/), and [Cron Triggers](https://developers.cloudflare.com/workers/platform/cron-triggers/) require a custom entrypoint file. Because the `@astrojs/cloudflare` adapter generates the entrypoint used for serving a site, these features have previously not been possible to use with Astro. Astro 5.10 adds support for customizing the entrypoint file used by the `@astrojs/cloudflare` adapter, giving you full control over your Cloudflare Workers setup.

Author: We generally include all the adapter minor updates, as well as experimental feature updates that went out in patches in the minor post and treat them as if they're all one update. I might change this to just say "this release" though.

To customize the entrypoint, add the `workerEntryPoint` option to your `@astrojs/cloudflare` adapter configuration in your `astro.config.mjs` file:

```js title="astro.config.mjs" {6-9}
import { defineConfig } from 'astro/config';
import cloudflare from '@astrojs/cloudflare';

export default defineConfig({
	adapter: cloudflare({
		workerEntryPoint: {
			path: 'src/worker.ts',
			namedExports: ['MyDurableObject']
		}
	}),
});
```

You can then create a custom entrypoint file at the specified path, which can include any additional imports or exports you need for your Workers setup. This example configures a custom entry file that registers a Durable Object and a queue handler:

```ts title="src/worker.ts" "handle(manifest, app, request, env, ctx)"
import type { SSRManifest } from 'astro';

import { App } from 'astro/app';
import { handle } from '@astrojs/cloudflare/handler'
import { DurableObject } from 'cloudflare:workers';

class MyDurableObject extends DurableObject<Env> {
	constructor(ctx: DurableObjectState, env: Env) {
		super(ctx, env)
	}
}

> 

Reviewer 4: Do we actually want to use a more practical example here, or do we trust users that want to use DurableObjects know how they work, since it is an advanced use-case anyways?

Author: I just used the example from your changelog, which I think is fine. It's just showing how it's done.

export function createExports(manifest: SSRManifest) {
	const app = new App(manifest);
	return {
		default: {
			async fetch(request, env, ctx) {

> 				await env.MY_QUEUE.send("log");

Reviewer 4: I think this is obvious, but to make this work. A user must configure the proper Bindings in either `wrangler.toml` or `wrangler.jsonc`.
